php_phongo.c

```diff
@@ -1634,7 +1634,6 @@ static void php_phongo_cursor_iterator_rewind(zend_object_iterator *iter TSRMLS_
 	php_phongo_cursor_iterator *cursor_it = (php_phongo_cursor_iterator *)iter;
 	php_phongo_cursor_t        *cursor = cursor_it->cursor;
 
-	php_phongo_cursor_free_current(cursor);
```

Comment:
Turns out you were somewhat right.

These free_current() are all harmless, so I left them -- but I managed to reproduce the issue of iterating over a cursor twice -- where the second time we would only access the cursor->firstBatch document.